lib/msbuild-finder.js

```diff
@@ -57,13 +57,25 @@ module.exports.find = function (options) {
     throw new PluginError(constants.PLUGIN_NAME, 'No MSBuild Version was supplied!');
   }
 
-  if (version === '12.0' || version === '14.0') {
+  if (version === '12.0' || version === '14.0' || version === '15.0') {
     // On 64-bit systems msbuild is always under the x86 directory. If this
     // doesn't exist we are on a 32-bit system. See also:
     // https://blogs.msdn.microsoft.com/visualstudio/2013/07/24/msbuild-is-now-part-of-visual-studio/
     var pathRoot = env_var_dir || path.join('C:', is64Bit ? 'Program Files (x86)' : 'Program Files');
     var x64_dir = is64Bit ? 'amd64' : '';
-    return path.join(pathRoot, 'MSBuild', version, 'Bin', x64_dir, 'MSBuild.exe');
+    var msbuildRoot = pathRoot
+
+    if(version === '15.0')
```

Comment:
I'd misunderstood this, Visual Studio defaults to install at "\Program Files (x86)\Microsoft Visual Studio\2017" with a subfolder of Enterprise, Professional, Community or BuildTools depending on which flavour(s) you've installed.  MsBuild looks for adjacent targets which are more complete in the more expensive installations. It makes sense to check all of these folders.  I'll leave it dropping back to MsBuild\15.0\Bin, but that can never exist as far as I can see.